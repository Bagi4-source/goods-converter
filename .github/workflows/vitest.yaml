name: 'Test'
on:
  pull_request:
  push:

jobs:
  vitest:
    name: Run vitest
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
      - uses: actions/checkout@v2
      - name: 'Install Node'
        uses: actions/setup-node@v2
        with:
          node-version: '18.x'
      - name: 'Install Deps'
        run: npm i
      - name: 'Test'
        run: npm run test
      - name: 'Coverage'
        run: npm run coverage
      - name: 'Report Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2