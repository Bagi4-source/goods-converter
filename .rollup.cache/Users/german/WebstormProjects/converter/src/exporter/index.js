export * from "./exporter";
export * from "./exporter.types";
//# sourceMappingURL=index.js.map