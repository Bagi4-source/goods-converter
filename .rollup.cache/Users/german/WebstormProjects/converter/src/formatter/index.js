export * from "./CSV.formatter";
export * from "./YML.formatter";
export * from "./formater.types";
//# sourceMappingURL=index.js.map