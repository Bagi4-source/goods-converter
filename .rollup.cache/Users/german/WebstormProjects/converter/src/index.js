export * from "./exporter";
export * from "./formatter";
//# sourceMappingURL=index.js.map