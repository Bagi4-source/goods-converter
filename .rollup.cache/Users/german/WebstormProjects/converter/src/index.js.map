{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,YAAY,CAAA;AAC1B,cAAc,aAAa,CAAA","sourcesContent":["export * from \"./exporter\"\nexport * from \"./formatter\""]}