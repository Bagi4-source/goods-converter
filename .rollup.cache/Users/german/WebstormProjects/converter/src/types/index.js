export * from "./Product.types";
export * from "./Category.types";
//# sourceMappingURL=index.js.map