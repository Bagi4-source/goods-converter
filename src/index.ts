export * from "./exporter"
export * from "./formatter"
export * from "./types"