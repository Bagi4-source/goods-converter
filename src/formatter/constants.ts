import {CSVFormatter} from "./CSV.formatter";
import {YMLFormatter} from "./YML.formatter";
import {TildaFormatter} from "./Tilda.formatter";

export const Formatters = {
    CSV: new CSVFormatter(),
    YML: new YMLFormatter(),
    Tilda: new TildaFormatter(),
}
