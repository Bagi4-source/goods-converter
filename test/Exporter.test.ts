import { describe, expect, it, vi } from "vitest";

import { Formatters, GoodsExporter } from "../src";
import { categories, products } from "./constants";

describe("GoodsExporter", () => {
  vi.useFakeTimers().setSystemTime(new Date("2020-01-01"));

  const exporter = new GoodsExporter();

  it("check export", async () => {
    const data = await exporter.export(products, categories);

    expect(data.toString("utf-8")).toMatchSnapshot();
  });

  it("check export with transformers", async () => {
    exporter.setFormatter(new Formatters.YMLFormatter());
    exporter.setExporter((data: Buffer) => {
      return data;
    });
    exporter.setTransformers([
      (product) => ({
        ...product,
        price: product.price + 10000,
        images: product.images?.map((image) => image.replace("image", "pic")),
      }),
    ]);

    const data = await exporter.export(products, categories);

    expect(data.toString("utf-8")).toMatchSnapshot();
  });
  it("check export without transformers", async () => {
    exporter.setFormatter(new Formatters.ExcelFormatter());

    const data = await exporter.export(products, categories);

    expect(data).toMatchSnapshot();
  });
});
