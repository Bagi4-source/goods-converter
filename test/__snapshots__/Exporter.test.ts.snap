// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GoodsExporter > check export with formatter 1`] = `
"url;productId;parentId;variantId;title;description;vendor;vendorCode;category;images;videos;timeDeliveryMin;timeDeliveryMax;price;oldPrice;purchasePrice;currency;saleDate;countryOfOrigin;tags;codesTN;params;properties;sizes;keywords;relatedProducts;vendorId;categoryId;additionalExpenses;minQuantity;stepQuantity;available;barcode;weight;boxCount;dimensions;vat;count;favoriteCount;age;manufacturerWarranty;certificate;seriesName;gender
http://localhost:8080/?param1=value1&param2=value2;1;;1111;Title;Description;Nike;NIKE-1111;Обувь;image1,image2;;;;19000;20000;15000;RUR;01.01.2020;Китай;Nike,Кроссовки;code;param1=v1, param2=v2;prop1=v1, prop2=v2;UK=36,37,38,39;Обувь,Кроссовки;1234;2;1;1000;1;1;true;567890567893;0.15;1;12/32/43;VAT_20;24;314;[object Object];true;111;Jordan 4;Мужское
;1;1111;1112;Title;Description;Nike;NIKE-1111;Обувь;image1,image2;;;;19000;20000;15000;RUR;01.01.2020;Китай;Nike,Кроссовки;code;size=44, color=white;seasons=spring, summer;;;;2;1;1000;1;1;true;567890567893;0.15;1;12/32/43;VAT_20;24;900;[object Object];true;111;Jordan 4;
"
`;

exports[`GoodsExporter > check export with transformers 1`] = `
"<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<yml_catalog date="2023-01-01T00:00:00">
<shop>
<categories>
  <category id="1" parentId="2">Обувь</category>
  <category id="2" parentId="3">Одежда, обувь и аксессуары</category>
  <category id="3" parentId="">Все товары</category>
</categories>

<offers>
<offer id="1111">
  <name>Title</name>
  <price>29000</price>
  <oldprice>20000</oldprice>
  <purchase_price>15000</purchase_price>
  <additional_expenses>1000</additional_expenses>
  <cofinance_price>0</cofinance_price>
  <currencyId>RUR</currencyId>
  <categoryId>1</categoryId>
  <vendorId>2</vendorId>
  <vendor>Nike</vendor>
  <vendorCode>NIKE-1111</vendorCode>
  <picture>pic1</picture>
  <picture>pic2</picture>
  <available>true</available>
  <series>Jordan 4</series>
  <min-quantity>1</min-quantity>
  <step-quantity>1</step-quantity>
  <size name="UK" delimiter=",">36,37,38,39</size>
  <keyword>Обувь</keyword>
  <keyword>Кроссовки</keyword>
  <saleDate>01.01.2020</saleDate>
  <property name="prop1">v1</property>
  <property name="prop2">v2</property>
  <param name="param1">v1</param>
  <param name="param2">v2</param>
  <description>
    <![CDATA[Description]]>
  </description>
  <country_of_origin>Китай</country_of_origin>
  <barcode>567890567893</barcode>
  <vat>VAT_20</vat>
  <count>24</count>
  <set-ids>Nike, Кроссовки</set-ids>
  <adult>false</adult>
  <downloadable>false</downloadable>
  <period-of-validity-days></period-of-validity-days>
  <comment-validity-days></comment-validity-days>
  <service-life-days></service-life-days>
  <comment-life-days></comment-life-days>
  <warranty-days></warranty-days>
  <comment-warranty></comment-warranty>
  <manufacturer_warranty>true</manufacturer_warranty>
  <certificate>111</certificate>
  <url>http://localhost:8080/?param1=value1&amp;param2=value2</url>
  <weight>0.15</weight>
  <dimensions>12/32/43</dimensions>
  <boxCount>1</boxCount>
  <disabled>false</disabled>
  <age unit="year">6</age>
  <tn-ved-codes>
    <tn-ved-code>code</tn-ved-code>
  </tn-ved-codes>
  <relatedProduct>1234</relatedProduct>
  <gender>Мужское</gender>
</offer>

<offer id="1112" group_id="1111">
  <name>Title</name>
  <price>29000</price>
  <oldprice>20000</oldprice>
  <purchase_price>15000</purchase_price>
  <additional_expenses>1000</additional_expenses>
  <cofinance_price>0</cofinance_price>
  <currencyId>RUR</currencyId>
  <categoryId>1</categoryId>
  <vendorId>2</vendorId>
  <vendor>Nike</vendor>
  <vendorCode>NIKE-1111</vendorCode>
  <picture>pic1</picture>
  <picture>pic2</picture>
  <available>true</available>
  <series>Jordan 4</series>
  <min-quantity>1</min-quantity>
  <step-quantity>1</step-quantity>
  <saleDate>01.01.2020</saleDate>
  <property name="seasons">spring, summer</property>
  <param name="size">44</param>
  <param name="color">white</param>
  <description>
    <![CDATA[Description]]>
  </description>
  <country_of_origin>Китай</country_of_origin>
  <barcode>567890567893</barcode>
  <vat>VAT_20</vat>
  <count>24</count>
  <set-ids>Nike, Кроссовки</set-ids>
  <adult>false</adult>
  <downloadable>false</downloadable>
  <period-of-validity-days></period-of-validity-days>
  <comment-validity-days></comment-validity-days>
  <service-life-days></service-life-days>
  <comment-life-days></comment-life-days>
  <warranty-days></warranty-days>
  <comment-warranty></comment-warranty>
  <manufacturer_warranty>true</manufacturer_warranty>
  <certificate>111</certificate>
  <weight>0.15</weight>
  <dimensions>12/32/43</dimensions>
  <boxCount>1</boxCount>
  <disabled>false</disabled>
  <age unit="year">6</age>
  <tn-ved-codes>
    <tn-ved-code>code</tn-ved-code>
  </tn-ved-codes>
</offer>

</offers>
</shop>
</yml_catalog>
"
`;

exports[`GoodsExporter > check export without transformers 1`] = `
"url;productId;parentId;variantId;title;description;vendor;vendorCode;category;images;videos;timeDeliveryMin;timeDeliveryMax;price;oldPrice;purchasePrice;currency;saleDate;countryOfOrigin;tags;codesTN;params;properties;sizes;keywords;relatedProducts;vendorId;categoryId;additionalExpenses;minQuantity;stepQuantity;available;barcode;weight;boxCount;dimensions;vat;count;favoriteCount;age;manufacturerWarranty;certificate;seriesName;gender
http://localhost:8080/?param1=value1&param2=value2;1;;1111;Title;Description;Nike;NIKE-1111;Обувь;pic1,pic2;;;;29000;20000;15000;RUR;01.01.2020;Китай;Nike,Кроссовки;code;param1=v1, param2=v2;prop1=v1, prop2=v2;UK=36,37,38,39;Обувь,Кроссовки;1234;2;1;1000;1;1;true;567890567893;0.15;1;12/32/43;VAT_20;24;314;[object Object];true;111;Jordan 4;Мужское
;1;1111;1112;Title;Description;Nike;NIKE-1111;Обувь;pic1,pic2;;;;29000;20000;15000;RUR;01.01.2020;Китай;Nike,Кроссовки;code;size=44, color=white;seasons=spring, summer;;;;2;1;1000;1;1;true;567890567893;0.15;1;12/32/43;VAT_20;24;900;[object Object];true;111;Jordan 4;
"
`;
